extensions/azure/vault/src/test/java/org/eclipse/dataspaceconnector/core/security/azure/AzureVaultExtensionTest.java

```diff
@@ -22,13 +22,13 @@
 import org.mockito.MockedStatic;
 
 import java.util.UUID;
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
 
-import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.atLeastOnce;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 class AzureVaultExtensionTest {
```

Comment:
I don't think the tests in this class are actually async, since they test the `initialize` method, that isn't doing any async test, in fact the assertions/verifications you added assert nothing, that's why test passes.

I think you can remove the `when` part and verify instead that the static method is actually called through a `verify`.

(as a sidenote, I think that static mocking highlights a bad design, and this design is actually bad, but I would leave it as it is in this PR, just make the tests actually verify what they needs)